### DISCUSSION TITLE
Sending a custom header when getting an SSE Realtime ClientID?

### DISCUSSION THREAD

**Discussion by AfdhalAtiffTan** [pocketbase/pocketbase#4554] [Q&A] [answered]
Hello,

I am working on a reverse proxy server for two Pocketbase instances (RAM and ROM) and would like allow it to request an SSE ClientID depending on the header like this:

```js
pb.collection("my_collection").subscribe("*", async (evt) => {
    console.log(evt);
}, {
    headers: { myHeaderKey: "myHeaderValue" }
});
```

I find that the header is not being sent so I tried the following:
```js
pb.beforeSend = function (url, options) {
    options.headers = Object.assign({}, options.headers, {
        'myHeaderKey': "myHeaderValue",
    });
    return { url, options };
};
```

The header isn't sent when with `GET /api/realtime`, it only worked with `POST /api/realtime`:

Is it possible to set header for  `GET /api/realtime`?

Thank you.

**Comment by ganigeorgiev** [comment]
I don't fully understand your use case but the `beforeSend` hook is fired only when sending regular requests (aka. when the request is send with `pb.send()`, which uses `fetch` under the hood).

For establishing SSE connections the JS SDK uses the native [`EventSource` interface](https://developer.mozilla.org/en-US/docs/Web/API/EventSource) and it doesn't support sending custom headers with the initial connect request (_and browser/html-spec developers don't seem to be interested adding it anytime soon - https://github.com/whatwg/html/issues/2177_). This is also the reason why the PocketBase realtime authentication happens at later stage, during the subscription.

**Reply by ganigeorgiev** [reply, marked as answer]
_Note that there were plans to remove the `EventSource` entirely and replace it with raw `fetch` stream but I had some issues with React Native and its partial `fetch` implementation, so this is left aside for now._

**Reply by AfdhalAtiffTan** [reply]
Hello @ganigeorgiev, thank you for the very quick reply!

I was trying to use the proxy server for emulating table stored in two places, e.g. ram_collection and rom_collection. 
The goal was to use the same endpoint port number for multiple database location.
At the moment I am using the collection name as the router, works fine so far apart from the SSE.

Thank you for the links, they are very helpful.
